sdk/storage/Azure.Storage.Common/tests/PooledMemoryStreamTests.cs

```diff
@@ -105,6 +105,32 @@ public async Task WriteStream(int dataSize, int bufferPartitionSize)
             Assert.AreEqual(0, pooledMemoryStream.Position);
         }
 
+        [TestCase(1, 0, 1)]
+        [TestCase(Constants.KB, 512, 2 * Constants.KB)]
+        [TestCase(Constants.KB, 512, 512)]
+        public async Task ReadByte(int dataSize, int initialReadSize, int bufferPartitionSize)
```

Comment:
insert some cases with non powers of two.
Also maybe add a test where you loop readbyte over the entire stream and put in in an array and see if you get the expected sequence of bytes.